packages/ariakit-utils/src/__tests__/dom-test.ts

```diff
@@ -0,0 +1,601 @@
+import {
+  closest,
+  contains,
+  getActiveElement,
+  getDocument,
+  getPopupRole,
+  getScrollingElement,
+  getTextboxSelection,
+  getWindow,
+  isButton,
+  isFrame,
+  isPartiallyHidden,
+  isTextField,
+  isVisible,
+  matches,
+  scrollIntoViewIfNeeded,
+} from "../dom";
+
+test("getDocument", () => {
+  const initialInnerHTML = document.body.innerHTML;
+  document.body.innerHTML = '<div id="testNode"></div>';
+  const node = document.getElementById("testNode");
+
+  expect(getDocument(undefined)).toEqual(document);
+  expect(getDocument(null)).toEqual(document);
+  expect(getDocument(node)).toEqual(document);
+
+  // Reset
+  document.body.innerHTML = initialInnerHTML;
+});
+
+test("getWindow", () => {
+  const initialInnerHTML = document.body.innerHTML;
+  document.body.innerHTML = '<div id="testNode"></div>';
+  const node = document.getElementById("testNode");
+
+  expect(getWindow(undefined)).toEqual(window);
+  expect(getWindow(null)).toEqual(window);
+  expect(getWindow(node)).toEqual(window);
+
+  // Reset
+  document.body.innerHTML = initialInnerHTML;
+});
+
+test("getActiveElement", () => {
+  const initialInnerHTML = document.body.innerHTML;
+  document.body.innerHTML = `
+    <div id="testNode">
+      <input id="testInput" />
+      <input type="text" aria-activedescendant="cb1-opt6" aria-readonly="true" aria-owns="cb1-list" aria-autocomplete="list" role="combobox" id="cb1-edit"/>
+      <ul aria-expanded="true" role="listbox" id="cb1-list">
+        <li role="option" id="cb1-opt1">Alabama</li>
+        <li role="option" id="cb1-opt2">Alaska</li>
+        <li role="option" id="cb1-opt3">American Samoa</li>
+        <li role="option" id="cb1-opt4">Arizona</li>
+        <li role="option" id="cb1-opt5">Arkansas</li>
+        <li role="option" id="cb1-opt6">California</li>
+        <li role="option" id="cb1-opt7">Colorado</li>
+      </ul>
+    </div>
+  `;
+  const node = document.getElementById("testNode");
+
+  // Test before any elemeted is focused
+  expect(getActiveElement(undefined)).toEqual(document.body);
+  expect(getActiveElement(null)).toEqual(document.body);
+  expect(getActiveElement(node)).toEqual(document.body);
+
+  // Test with an element focused
+  const input = document.getElementById("testInput");
+  input?.focus();
+
+  expect(getActiveElement(undefined)).toEqual(input);
+  expect(getActiveElement(null)).toEqual(input);
+  expect(getActiveElement(node)).toEqual(input);
+  expect(getActiveElement(input)).toEqual(input);
+
+  // Test IE11 Case
+  const activeElementMock = jest
+    .spyOn(document, "activeElement", "get")
+    .mockReturnValue(null);
+
+  expect(getActiveElement(undefined)).toEqual(null);
+  expect(getActiveElement(null)).toEqual(null);
+  expect(getActiveElement(node)).toEqual(null);
+  expect(getActiveElement(input)).toEqual(null);
```

Comment:
I'm not sure `toBe` works with `null`. But let's use it whenever possible.
```suggestion
  expect(getActiveElement(undefined)).toBe(null);
  expect(getActiveElement(null)).toBe(null);
  expect(getActiveElement(node)).toBe(null);
  expect(getActiveElement(input)).toBe(null);
```